worker/firewaller/firewaller.go

```diff
@@ -1160,6 +1242,36 @@ func (fw *Firewaller) forgetUnit(unitd *unitData) {
 	}
 }
 
+func (fw *Firewaller) configuredModelFirewall() (firewall.IngressRules, error) {
+	cfg, err := fw.firewallerApi.ModelConfig()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	// We don't need to watch controller config because APIPort is immutable
+	ctrlCfg, err := fw.firewallerApi.ControllerConfig()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	isController, err := fw.firewallerApi.IsControllerModel()
```

Comment:
We do not want to do it this way. This is an artefact of the fact that we are incorrectly exposing the implementation as model config detail. We want 2 apis:
- a model firewall rules change watcher
- a model firewall rules getter

Then all this worker has to do is fetch the rules info when something changes. The server side determines what needs to be served up based on controller model etc. The responsibility of this worker is simply to open and close ports according to what it is told.